homeassistant/components/smartenergy_goecharger/config_flow.py

```diff
@@ -0,0 +1,185 @@
+"""go-e Charger Cloud config flow and options flow setup."""
+
+import logging
+import re
+from typing import Any, Literal
+
+from goechargerv2.goecharger import GoeChargerApi
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import CONF_API_TOKEN, CONF_HOST, CONF_NAME, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class InvalidAuth(HomeAssistantError):
+    """Error to indicate there is invalid auth."""
+
+
+def _get_config_values(data_input: dict) -> dict:
+    data: dict = {}
+    config: list[str] = [CONF_NAME, CONF_HOST, CONF_API_TOKEN, CONF_SCAN_INTERVAL]
+
+    for config_name in config:
+        data[config_name] = data_input.get(config_name)
+
+    return data
```

Comment:
This confuses me. You are taking a dict, then making a new dict with the same keys and values as the original dict? Why can't you use the user_input dict? Why do you need to make a copy